### PR TITLE
Add documentation for update ecs service

### PR DESCRIPTION
**Pull request by tekerson**

### CODE REVIEW

File: docs/deployments/aws/ecs-update-service/index.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+title: Update Amazon ECS Service
+description: Update an existing Amazon ECS Service.
+---
+
+Octopus supports deploying a new release to an existing ECS Service through the `Update Amazon ECS Service` step.
+This step provides an opinionated deployment workflow that allows new released to be deployed to an ECS Cluster that is managed externally, for example manually or via Terraform.
+
+:::hint
+The `Update Amazon ECS Step` was added in Octopus **2021.3**. Presently only **Fargate** clusters are supported.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:10]:
```suggestion
The `Update Amazon ECS Service` step was added in Octopus **2021.3**. Presently only **Fargate** clusters are supported.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+---
+title: Update Amazon ECS Service
+description: Update an existing Amazon ECS Service.
+---
+
+Octopus supports deploying a new release to an existing ECS Service through the `Update Amazon ECS Service` step.
+This step provides an opinionated deployment workflow that allows new released to be deployed to an ECS Cluster that is managed externally, for example manually or via Terraform.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:7]:
```suggestion
This step provides an opinionated deployment workflow that allows new releases to be deployed to an ECS Cluster that is managed externally, for example manually or via Terraform.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+:::
+
+At a high level, the `Update Amazon ECS Service` step will:
+
+* Select the Docker image tags for the task definition (version selection is performed when creating a release).
+* Fetch an existing Task Definition to use as a template.
+* Create a new Task Definition revision based on the template definition and the step configuration provided.
+* Update the Service to use the newly created Task Definition.
+
+The following instructions can be followed to configure the `Update Amazon ECS Service` step.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:20]:
```suggestion
The following instructions can be used to configure the `Update Amazon ECS Service` step.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+:::hint
+The benefits of using deployment targets for ECS are outlined in the [ECS RFC blog post](https://octopus.com/blog/rfc-ecs-integration-with-octopus#why-use-targets).
+:::
+
+## Step 3: Add the Update ECS Service step
+
+Add the `Update Amazon ECS Service` step to the project, and provide it a name.
+
+As the step is using a deployment target a target role will also need to be specified.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:50]:
```suggestion
As the step is using a deployment target, a target role will also need to be specified.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+The role will be used to determine which ECS cluster to deploy to.
+Use the same role that you applied to your deployment target in Step 2.
+
+![Update ECS Step General Settings](images/update-ecs-general-settings.png "width=500")
+
+### Configuration section
+
+Specify the name of your Destination Task Definition.
+This is the task definition that will receive a new revision upon deploy.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:59]:
```suggestion
This is the task definition that will receive a new revision upon deployment.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+![Update ECS Step Configuration Section](images/update-ecs-configuration.png "width=500")
+
+### Container Definitions section
+
+At least one container definition must be specified when updating a task definition.
+
+![Update ECS Step Container Definitions Section](images/update-ecs-container-definitions.png "width=500")
+
+Specify the container name that will be used to locate the container definition within your task and select a feed and image that will be run by your task.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:76]:
```suggestion
Specify the container name that will be used to locate the container definition within your task and select a feed and image that your task will run.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+The specific image version will be specified later, when creating a release.
+
+![ECS Step Container Definition Parameters](images/update-ecs-container-definition.png "width=500")
+
+#### Environment Files
+
+List up to 10 environment files to be added to the container.
+Environment files should be stored in S3, they must have a ".env" extension, and each line should contain a "VARIABLE=VALUE" variable definition.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:84]:
```suggestion
Environment files should be stored in S3, they must have a ".env" extension, and each line should contain a `VARIABLE=VALUE` variable definition.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+The configured environment file list can be used to either completely replace or be appended to the container's existing environment file list.
+
+Each environment file has a type to configure the storage device (S3 is currently the only option) and a source to specify the location within that storage (for S3, this is an ARN).
+
+![Update ECS Step Container Definition Environment Files](images/update-ecs-environment-files.png "width=500")
+
+#### Environment Variables
+
+You can add either plain text or secret environment variables to the container.
+Plain text environment variables will use the value provided, while secret variables refer to a secret value stored in the AWS Secrets Manager.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:95]:
```suggestion
Plain text environment variables will use the value provided, while secret variables refer to a secret value stored in AWS Secrets Manager.
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+Selecting the `Don't wait` option means that the step will not fail if the tasks fail to properly start.
+:::
+
+#### Variable replacements
+
+Any of the input fields can be bound to an Octostache expression. [Variable substitution](/docs/projects/variables/variable-substitutions.md) will be performed before the release is deployed.
+
+### Output variables
+
+Presently, the step does outputs the following variables.
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:125]:
```suggestion
The step outputs the following variables:
```

File: docs/deployments/aws/ecs-update-service/index.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+TaskDefinitionRevision
```
Comment (harrisonmeister) [inline docs/deployments/aws/ecs-update-service/index.md:127]: I think these variables would be better placed in a table. Something like this:

```
| Name | Description |
| -------------------- | -------|
|`TaskDefinitionRevision` | The revision number of the newly created task (if a new revision was created) |
|`TaskDefinitionFamily` | The name of a family that the new task definition is registered to (if a new revision was created) |
|`ClusterName` | The name of the cluster to which the updated service belongs |
|`Region` | The AWS region in which the operations were executed |
```
That would output like this:

| Name | Description |
| -------------------- | -------|
|`TaskDefinitionRevision` | The revision number of the newly created task (if a new revision was created) |
|`TaskDefinitionFamily` | The name of a family that the new task definition is registered to (if a new revision was created) |
|`ClusterName` | The name of the cluster to which the updated service belongs |
|`Region` | The AWS region in which the operations were executed |

**Review by harrisonmeister — COMMENTED** [review]
Made some minor line edit suggestions

File: docs/deployments/aws/ecs-update-service/index.md (hunk 10 above)
Comment (tekerson) [reply docs/deployments/aws/ecs-update-service/index.md:127]: Yep, I did consider making it a table but I didn't know how people felt about them. The format I used is supposed to be a definition list but it's an extended markdown syntax and I wasn't sure if our parser/renderer supports it.

I'll change it for a table.

File: docs/deployments/aws/ecs-update-service/index.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+: The name of a family that the new task definition is registered to (if a new revision was created)
+
+ClusterName
+: The name of the cluster to which the updated service belongs
+
+ServiceName
+: The name of the updated service
+
+Region
+: The AWS region in which the operations were executed
```
Comment (tekerson) [inline docs/deployments/aws/ecs-update-service/index.md:140]:
```suggestion
| Name | Description |
| -------------------- | -------|
|`TaskDefinitionRevision` | The revision number of the newly created task (if a new revision was created) |
|`TaskDefinitionFamily` | The name of a family that the new task definition is registered to (if a new revision was created) |
|`ClusterName` | The name of the cluster to which the updated service belongs |
|`Region` | The AWS region in which the operations were executed |
```

**Review by harrisonmeister — APPROVED** [review]
LGTM

**Comment by harrisonmeister** [comment]
@tekerson - I'm happy to merge, or if you want to that's cool. I wasn't sure if you were making any additional changes :+1:

**Comment by tekerson** [comment]
> @tekerson - I'm happy to merge, or if you want to that's cool. I wasn't sure if you were making any additional changes 👍

@harrisonmeister - No, not waiting on any further changes from my side but the feature isn't finalised/released yet so I was holding off on merging the docs until it is.

**Comment by egorpavlikhin** [comment]
@harrisonmeister the step is now in master, could you please merge the PR if you are happy with the recent changes?

**Pull request merged**